src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -457,6 +460,13 @@ private unsafe bool WasEventDetected(int pollFileDescriptor, int valueFileDescri
                 int waitResult = Interop.epoll_wait(pollFileDescriptor, out epoll_event events, 1, PollingTimeout);
                 if (waitResult == -1)
                 {
+                    var errorCode = Marshal.GetLastWin32Error();
+                    if (errorCode == ERROR_CODE_EINTR)
+                    {
+                        // ignore Interrupted system call error and retry
+                        continue;
+                    }
+
                     throw new IOException("Error while waiting for pin interrupts.");
```

Comment:
might be a good idea to add error code in this message as well for easier debugging

Reply:
My idea is to change it to use `ExceptionHelper` in a follow-up PR.

Reply:
I see @pgrawehr already asked for that. I'd prefer we changed it in this PR, it's not much different concern. I'd just add append something like ` (ErrorCode={ERROR_CODE_EINTR})` and change string to interpolated. We can create another issue to have more uniform way of throwing but it's better to improve it now as we're touching this code

Reply:
if you're planning to send a follow up then that's fine too but just want to make sure we actually do that - usually it's simpler to just do it in a single PR, also we can do the temporary fix now and follow up better fix in case you forget to send follow-up 😄 (sorry but happens to everyone for various reasons)

Reply:
Added error code to message as requested.